core/arch/arm/kernel/secure_partition.c

```diff
@@ -422,6 +379,196 @@ static TEE_Result sp_dt_get_uuid(const void *fdt, int node,
 	return TEE_SUCCESS;
 }
 
+static TEE_Result sp_is_elf_format(const void *fdt, int sp_node,
+				   bool *is_elf_format)
+{
+	TEE_Result res = TEE_SUCCESS;
+	uint32_t elf_format = 0;
+
+	res = sp_dt_get_u32(fdt, sp_node, "legacy-elf-format", &elf_format);
+	if (res != TEE_SUCCESS && res != TEE_ERROR_ITEM_NOT_FOUND)
+		return res;
+
+	*is_elf_format = (elf_format != 0);
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result sp_binary_open(const TEE_UUID *uuid,
+				 const struct ts_store_ops **ops,
+				 struct ts_store_handle **handle)
+{
+	TEE_Result res = TEE_ERROR_ITEM_NOT_FOUND;
+
+	SCATTERED_ARRAY_FOREACH(*ops, sp_stores, struct ts_store_ops) {
+		res = (*ops)->open(uuid, handle);
+		if (res != TEE_ERROR_ITEM_NOT_FOUND &&
+		    res != TEE_ERROR_STORAGE_NOT_AVAILABLE)
+			break;
+	}
+
+	return res;
+}
+
+static TEE_Result load_binary_sp(struct ts_session *s,
+				 struct user_mode_ctx *uctx)
+{
+	size_t bin_size = 0, bin_size_rounded = 0, bin_page_count = 0;
+	const struct ts_store_ops *store_ops = NULL;
+	struct ts_store_handle *handle = NULL;
+	TEE_Result res = TEE_SUCCESS;
+	tee_mm_entry_t *mm = NULL;
+	struct mobj *mobj = NULL;
+	uaddr_t base_addr = 0;
+	uint32_t vm_flags = 0;
+	unsigned int idx = 0;
+	vaddr_t va = 0;
+
+	if (!s || !uctx)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	DMSG("Loading generic format SP %pUl", &uctx->ts_ctx->uuid);
```

Comment:
s/generic/raw binary/

Reply:
Done, also changed the commit message to follow the same naming convention.